### DISCUSSION TITLE
Can I enable typeguard for some tests only/exclude for some tests?

### DISCUSSION THREAD

**Discussion by MartinThoma** [agronholm/typeguard#255] [Q&A]
I am currently adding type annotations to PyPDF2. I'm running it like this:

```
$ pytest tests --typeguard-packages=PyPDF2
```

This is mostly fine, but there are a few tests that cause issues. I would like to deactivate typeguard for those. Is that possible?

If not, is it possible to activate typeguard only for a few tests of choice (e.g. in one test file)?

**Comment by agronholm** [comment]
You can suppress type checking with `with typeguard.suppress_type_checks(): ...` as needed. If you're using pytest, you could put that in a fixture that would be used by any tests that don't need it.
